codegen/src/main/java/software/amazon/awssdk/codegen/poet/auth/scheme/AuthSchemeSpecUtils.java

```diff
@@ -178,17 +182,24 @@ public List<AuthType> serviceDefaultAuthTypes() {
         return Collections.singletonList(intermediateModel.getMetadata().getAuthType());
     }
 
-    public List<AuthType> allServiceAuthTypes() {
-        Set<AuthType> result =
-            intermediateModel.getOperations()
-                             .values()
-                             .stream()
-                             .map(OperationModel::getAuth)
-                             .flatMap(List::stream)
-                             .collect(Collectors.toSet());
-        List<AuthType> modeled = intermediateModel.getMetadata().getAuth();
-        result.addAll(modeled);
-        return result.stream().sorted().collect(Collectors.toList());
+    public Set<Class<?>> allServiceConcreteAuthSchemeClasses() {
+        Set<Class<?>> result =
+            Stream.concat(intermediateModel.getOperations()
+                                           .values()
+                                           .stream()
+                                           .map(OperationModel::getAuth)
+                                           .flatMap(List::stream),
+                          intermediateModel.getMetadata().getAuth().stream())
+                  .map(AuthSchemeCodegenMetadata::fromAuthType)
+                  .map(AuthSchemeCodegenMetadata::authSchemeClass)
+                  .filter(Objects::nonNull)
+                  .collect(Collectors.toCollection(() -> new TreeSet<>(Comparator.comparing(Class::getSimpleName))));
+
+        if (generateEndpointBasedParams()) {
```

Comment:
readability wise, `useEndpointBasedAuthProvider()` seems better

Reply:
Moreover, I think we should rename this customization from `enableEndpointAuthSchemeParams` to something like `enableEndpointAuthSchemeResolution` - the scope is generalized v/s specific to params which is just the first thing we customized. And then maybe we don't need multiple methods in this Utils class that checks that flag.

Reply:
Maybe `enableSraEndpointAuthSchemeResolution` to convey this is the SRA part of the feature v/s Endpoint 2.0 itself.

Reply:
I will switch to use `useEndpointBasedAuthProvider()` but let's discuss the renaming/refactoring in a separated thread, I'm not sure about the proposed name or adding the SRA which might only make sense internally.

Reply:
separate PR sounds good